### ISSUE TITLE
Circles move in time when zooming WebGL plot with datetime axes

### ISSUE THREAD

**Issue by samaust** [bokeh/bokeh#3137]
The screenshots below are generated using Bokeh 0.10.0 release in Firefox 42.0. When zooming a WebGL plot with a datetime axes, the points move in time as zoom level changes. The code below comes from the documentation with  webgl=True added and line changed to circle.

```
import pandas as pd
from bokeh.plotting import figure, output_file, show

AAPL = pd.read_csv(
        "http://ichart.yahoo.com/table.csv?s=AAPL&a=0&b=1&c=2000&d=0&e=1&f=2010",
        parse_dates=['Date']
    )

output_file("datetime.html")

# create a new plot with a datetime axis type
p = figure(width=800, height=250, x_axis_type="datetime", webgl=True)

p.circle(AAPL['Date'], AAPL['Close'], color='navy', alpha=0.5)

show(p)
```
At normal zoom
![plot_1](https://cloud.githubusercontent.com/assets/10716165/11195528/583acc2a-8c80-11e5-9537-c34fbe27884c.jpg)

Zoomed in on 1/01
![plot_2](https://cloud.githubusercontent.com/assets/10716165/11195531/5a8fc3d6-8c80-11e5-95df-e0e158d06ecf.jpg)

A bit more zoomed in on 1/01
![plot_3](https://cloud.githubusercontent.com/assets/10716165/11195534/5c7c7f5e-8c80-11e5-94e0-6398a452a1a9.jpg)

Notice how the points moves horizontally around 1/01. On the second plot, a point was on the line and in the third plot, that same point is more to the left and it's the next point to the right that is closer to the line.

I've noticed the same thing with a personnal dataset with a different number of points and different length. The points jump as much as a day or two in both directions when changing zoom level.

**Comment by canavandl** [comment]
Hi @samaust 

This issue was resolved in PR #2956 and the fix should be available in master or the upcoming 0.11.0 release. Thanks for submitting an issue!

**Comment by fpliger** [comment]
@canavandl @damianavila from comments above I understand that this is closed... please reopen the issue if I'm wrong...
